canvas/gradient.go

```diff
@@ -0,0 +1,190 @@
+package canvas
+
+import (
+	"image"
+	"image/color"
+	"image/draw"
+	"math"
+
+	"fyne.io/fyne"
+)
+
+// Gradient describes a gradient between two colors
+type Gradient struct {
+	baseObject
+
+	Generator func(w, h int) image.Image
+
+	Translucency float64 // Set a translucency value > 0.0 to fade the raster
```

Comment:
This should not be needed as we control the colour of the gradient at both ends (with alpha)